### ISSUE TITLE
How can I use SemanticKITTI dataset. + Can I get label?

### ISSUE THREAD

**Issue by ppccww0201** [KPeng9510/MASS#3]
ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡFROM README.TXTㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ
Master branch is currently set up for SemanticKITTI dataset. Config file can be found in tools/cfgs/nuscenes_models/cbgs_pp_multihead.yaml, please follow the guidance of OpenPCDet to run the code.

Code for nuScenes dataset can be found in branch develop_nusc.

For SemanticKitti dataset, if you have get accessed to Kitti-odo and SemanticKitti dataset, please forward the access email from KITTI to [pengkunyu1013@gmail.com](mailto:pengkunyu1013@gmail.com), the download link for the dense and sparse top-view semantic segmentation label will be given.
ㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡㅡ
Read the branch part in README.TXT and there are two issues.

1. In README.TXT, the master branch is set up for SemanticKITTI dataset, so all I have to do is setting the path for SemanticKITTI dataset in /pcdet/datasets/dataset.py?

2. I think I need 'dense_label', 'accupancy', 'sparse_label' which I saw in /pcdet/datasets/dataset.py for using the dataset.
Are they included in the label download link? (I sent you email for label download link.)

**Comment by KPeng9510** [comment]
Hi,

Thank you for your question!

regarding your first question, yes, you should set the corresponding path
in dataset.py

regarding your second question, please try _compute_visibility function in
mapping.cpp under the mapping folder and compute the occupancy. The
occupancy map is not provided, you could also generate it according to the
paper while having a look at https://github.com/peiyunh/wysiwyg.git repo.

Best,

Kunyu

**Comment by ppccww0201** [comment]
Thank you for your kind reply. I'm gonna try like you said.
In addition, how can I get download link for label?

Just sending request email to pengkunyu1013@gmail.com is right way to get the download link?

Or should I prepare something else and send email to pengkunyu1013@gmail.com?

(I already sent request email to pengkunyu1013@gmail.com for download link.)

Thank you!

**Comment by KPeng9510** [comment]
Hi,

Could you send me an email again? sorry maybe your last email got lost and
I can not find it now. If you forward me the download agreement email by
the application for KITTI semantic dataset, I will send you the link.

Best,

Kunyu

**Comment by ppccww0201** [comment]
I sent you email again.

If you haven't received the email again, please leave a reply.

Thank you!

**Comment by KPeng9510** [comment]
Hi，I have sent you the download link via email

**Comment by ppccww0201** [comment]
Hi, I encountered new problem.

Traceback (most recent call last):
  File "/home/MASS/PythonProject/MASS/tools/train.py", line 6, in <module>
    from test import repeat_eval_ckpt
  File "/home/MASS/PythonProject/MASS/tools/test.py", line 13, in <module>
    from eval_utils import eval_utils
  File "/home/MASS/PythonProject/MASS/tools/eval_utils/eval_utils.py", line 9, in <module>
    from pcdet.models import load_data_to_gpu
  File "/home/MASS/PythonProject/MASS/pcdet/models/__init__.py", line 6, in <module>
    from .detectors import build_detector
  File "/home/MASS/PythonProject/MASS/pcdet/models/detectors/__init__.py", line 1, in <module>
    from .detector3d_template import Detector3DTemplate
  File "/home/MASS/PythonProject/MASS/pcdet/models/detectors/detector3d_template.py", line 6, in <module>
    from ...ops.iou3d_nms import iou3d_nms_utils
  File "/home/MASS/PythonProject/MASS/pcdet/ops/iou3d_nms/iou3d_nms_utils.py", line 9, in <module>
    from . import iou3d_nms_cuda
ImportError: /home/MASS/PythonProject/MASS/pcdet/ops/iou3d_nms/iou3d_nms_cuda.cpython-38-x86_64-linux-gnu.so: undefined symbol: _ZN3c106detail14torchCheckFailEPKcS2_jRKSs

Process finished with exit code 1

I tried hard to google it, but I couldn't find the right answer. Please help me out!

I use 'scrin/dev:latest' as base image.
cuda == 11.0
python == 3.8
pytorch == 1.7.1+cu110

**Comment by KPeng9510** [comment]
Hi,

I remember that I used PyTorch <= 1.5.0 and>=1.1, this issue is mostly
caused by an incorrect Pytorch version. You could also check the Q&A of
https://github.com/open-mmlab/OpenPCDet which is the base of my code.

Thanks!

**Comment by ppccww0201** [comment]
Thanks for your reply! One more question please!

Can you tell me what did you use as base image for spconv?

Thanks a lot~!!

Android용 Outlook<https://aka.ms/AAb9ysg> 다운로드
________________________________

**Comment by ppccww0201** [comment]
Thanks for your reply! One more question please!

Could you tell me what did you use as base image for spconv?

Thanks a lot~!!

Android용 Outlook<https://aka.ms/AAb9ysg> 다운로드
________________________________

**Comment by ppccww0201** [comment]
Thanks for your reply! One more question please!

Could you tell me what did you use as base image for spconv?

Thanks a lot~!!

**Comment by KPeng9510** [comment]
Hi,

Sorry for the late reply, I searched the environment in my server however
it may be deleted. I compiled spconv according to the instructions of the
OpenPCDet repository. Maybe you could try to follow their tutorials.

Best,
Kunyu

**Comment by ppccww0201** [comment]
![image](https://user-images.githubusercontent.com/91767741/201338410-c981f68c-90e9-4555-8637-d31482a4f083.png)

The following error occurred while running train.py: (First picture)
Following the error, it seems that dataset.py requires 2D png data.
However, the result of running visi.py is a one-dimensional bin file.
How can I get the visibility map as 2D png data?

**Comment by KPeng9510** [comment]
Hi,

You could try to change the program and load it directly with *.bin file
using numpy through np.fromfile and reshape it into the desired input shape
(the same with the shape when you calculate the visibility) or try to map
the visibility map between 0 -255 to get the png image for the input. The
second case can improve the I/O speed.

**Comment by ppccww0201** [comment]
Hi, I was able to run the program successfully. Thanks to you.

I have a question about the observability map.
I succeeded in creating an observity map using _compute_visibility function in
mapping.cpp, but when I checked it, it was different from what I expected.

![image](https://user-images.githubusercontent.com/91767741/202358084-22e3ce8d-1135-44ad-ba9e-e872c280eb27.png)
<left: 2D occupancy map from MASS paper; right: observity map I created>

I expected a map like the picture on the left, but the map I created the picture on the right.
In addition, almost all pixel values are zero (ex: 497,451/500,000)

I wonder if the observity map I made is wrong, or is it originally right?

p.s. The observity map is so sparse that the segmentation result is as follows.
![image](https://user-images.githubusercontent.com/91767741/202361737-3ddd2fed-b714-4f2a-b96d-d0ac5e4b8a8e.png)
